Assets/AnkrDemo/Scripts/WebGLConnectionController.cs

```diff
@@ -1,52 +1,124 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AnkrDemo.Scripts;
 using AnkrSDK.Data;
+using Cysharp.Threading.Tasks;
+using Newtonsoft.Json;
 using UnityEngine;
 
 namespace AnkrSDK.Examples.UseCases.WebGlLogin
 {
-
 	public class WebGLConnectionController : MonoBehaviour
 	{
 #if UNITY_WEBGL
-		[SerializeField]
-		private WebGl.WebGLConnect _webGlConnect;
+		[SerializeField] private WebGl.WebGLConnect _webGlConnect;
+
+		[SerializeField] private WebGLLoginPanelController _webGlLoginManager;
 
-		[SerializeField]
-		private WebGLLoginPanelController _webGlLoginManager;
+		[SerializeField] private WebGLHeaderWalletsPanel _webGlLoginViewer;
+
+		private Dictionary<string, bool> _walletsStatus;
+		private TaskCompletionSource<WalletsStatus> _completionSource;
 
 		private void Awake()
 		{
+			_completionSource = new TaskCompletionSource<WalletsStatus>();
+
 			_webGlConnect.OnNeedPanel += ActivatePanel;
-			_webGlConnect.OnConnect += ChangeLoginPanel;
+			_webGlConnect.OnConnect += ChangePanels;
 			_webGlLoginManager.NetworkChosen += OnNetworkChosen;
 			_webGlLoginManager.WalletChosen += OnWalletChosen;
+			_webGlLoginViewer.ConnectTo += OnConnect;
+		}
+
+		private async void Start()
```

Comment:
don't make start async it will make the game pause until this start finishes. 
Extract async method and .Forget() it